app/models/approval_request.rb

```diff
@@ -0,0 +1,6 @@
+class ApprovalRequest < ApplicationRecord
+  validates :workflow_ref, :presence => true
+  enum :state => [:undecided, :approved, :denied]
```

Comment:
Is `:undecided == 'an error case'`

Reply:
Yep as we discussed "undecided" is all the things!